src/components/Modals/NewCourse/CourseConflictModal.vue

```diff
@@ -170,28 +173,42 @@ export default defineComponent({
       const maxNumSelfChecks = Math.max(...this.numSelfChecksPerConflict);
       return index === 1 && maxNumSelfChecks > 0;
     },
-    // get the reqs in conflictReqIds that are in conflict, based on course with uniqueID
-    // self check requirements not in conflict with the other reqs will be excluded
-    getReqsInConflict(uniqueID: string | number, conflictReqIds: string[]): string[] {
-      const constraintViolations = getConstraintViolationsForSingleCourse(
-        { uniqueId: uniqueID },
-        conflictReqIds,
-        (reqA, reqB) =>
-          allowCourseDoubleCountingBetweenRequirements(
-            store.state.userRequirementsMap[reqA],
-            store.state.userRequirementsMap[reqB]
-          )
-      );
+    // determine if each selectable req in selectableReqIds is in conflict with the reqs in conflictReqIds,
+    // based on course with uniqueID.
+    // return the list of conflictReqIds + selectableReqIds in conflict.
+    getReqsInConflict(
+      uniqueID: string | number,
+      conflictReqIds: string[],
+      selectableReqIds: string[]
+    ): string[] {
+      const selectableReqIdsInConflict: string[] = [];
+      selectableReqIds.forEach(selectableReqId => {
+        const constraintViolations = getConstraintViolationsForSingleCourse(
+          { uniqueId: uniqueID },
+          [...conflictReqIds, selectableReqId],
+          (reqA, reqB) =>
+            allowCourseDoubleCountingBetweenRequirements(
+              store.state.userRequirementsMap[reqA],
+              store.state.userRequirementsMap[reqB]
+            )
+        );
 
-      const validConflicts = constraintViolations.courseToRequirementsInConstraintViolations.get(
-        this.selectedCourse.uniqueID
-      );
-      let firstConflict: string[] = [];
-      if (validConflicts) {
-        [firstConflict] = [...validConflicts];
-      }
+        const validConflicts = constraintViolations.courseToRequirementsInConstraintViolations.get(
+          this.selectedCourse.uniqueID
+        );
+
+        if (!validConflicts || validConflicts.size !== 1) {
+          return;
+        }
+        const [firstConflict] = [...validConflicts];
+
+        // if selectable req is not in conflict with conflictReqIds, it will be missing from the first and only conflict
+        if (validConflicts && firstConflict.includes(selectableReqId)) {
+          selectableReqIdsInConflict.push(selectableReqId);
+        }
```

Comment:
Honestly I would just refactor this whole thing to a function in `requirements-constraints-utils`. But that's not super important for right now

I still think `const [firstConflict] = [...validConflicts];` is hard to understand. Also, is `if (validConflicts)` necessary if you check for it earlier? Maybe you can do something like `constraintViolations.requirementsThatDoNotAllowDoubleCounting.has(selectableReqId)`

Reply:
Another thought: is it possible for selectable requirements to conflict with each other? So say you have
- requirement A
- requirement B
- selectable requirement C
- selectable requirement D

You would check {A, B, C} and then {A, B, D}, and add C or D individually to the list of requirement ids if there are potential conflicts. But we should add both C _and_ D if there is a conflict like {A, B, C} and {C, D}, right? If so, maybe this function needs a bit more thought